### DISCUSSION TITLE
Problem with svg to jpg conversion: missing part of the image

### DISCUSSION THREAD

**Discussion by newalderaan** [ImageMagick/ImageMagick#2543] [Help] [answered]
Hi,

I am using version: _ImageMagick 7.0.10-29 Q16 x64 2020-09-05_ and trying to convert a Svg image to a Jpg but part of the image is lost during the conversion for some of the Svg files (not all). The command I am using is the following:

magick convert -density 1000  original_img.svg -resize 100% converted_img.jpg

Check attached images in zip file (contains only the Svg original one and the converted version), the missing part of the image is on the top-left side.

Thanks in advance.

[images.zip](https://github.com/ImageMagick/ImageMagick/files/5200491/images.zip)

**Comment by fmw42** [comment]
This command works fine for me on IM 7.0.10.29 Q16 Mac OSX when Imagemgick is using Inkscape to render the SVG. Note for IM 7, use magick, not magick convert and not convert.

`magick -density 1000 original_img.svg -resize 100% x.jpg`


![x](https://user-images.githubusercontent.com/13458218/92766907-7ccb0c80-f34b-11ea-90c1-05a409070948.jpg)

I suspect you are using the internal MSVG/XML renderer. You will get better results with the RSVG delegate and probably even better with Inkscape. If you put Inkscape on your system, you should not have to re-install ImageMagick and it will be used by default.

**Comment by newalderaan** [comment]
Hi,

Thank you very much for the response, really appreciate it. Looking at the image you posted I still see that something is missing in the top-left part of the image, we should see this full image (screenshot taken from a svg file viewer):

![image](https://user-images.githubusercontent.com/39128187/92924329-0a385a80-f439-11ea-80c4-6a93a1f3f5af.png)

Inkscape was already installed in my Windows 10 system when I executed the command, in addition the "bin" directory is included in the "path" environment variable (don't know if it makes any difference). Anyway, I have tested again using your version of the command but still have the same issue (same as your image above). I also tried the past day to use directly Inkscape but once again I got the same result (I'm afraid I don't remember the exact command I used).

> Inkscape 1.0.1 (3bc2e813f5, 2020-09-07)
>     Pango version: 1.46.1
> 

Any clue? To be honest, I'm a rookie with these tools, it is not something that I use frequently, it is more a one shot thing.

Thanks for your time.

**Comment by snibgo** [comment, marked as answer]
The problem is in your original_img.svg. I suggest you view it in Inkscape. You will see that the image extends to negative x-coordinates.

In Inkscape, open the "Export PNG window". You will see the page coordinates are from (0,0) to (348,281). But the drawing coordinates are from approx (-37,-9) to (385,290). The drawing is larger than the page.

In Inkscape, you can export the _drawing_ instead of the _page_. In batch, Inkscape has a "--export-area-drawing" option, which I suppose does what you want. See https://inkscape.org/doc/inkscape-man.html

Or in Inkscape, you can fix the page size with Document properties, Resize page to content, Resize page to drawing or selection.

I don't think you can make the adjustment from the IM command line. But you can edit the SVG, to:
```
viewBox="-37 -9 385 290"
```

**Comment by newalderaan** [comment]
Hi @snibgo ,

Thank you very much for your answer, as you said the problem were the coordinates and the -D argument did the trick, this is the small script I used to change a bunch of svg files, probably there are better ways but it worked for me.

> find . -name "*.svg" -print0 | while read -d $'\0' file
> do
>   cat $file | inkscape -p -D --export-dpi=300 --export-type=png | magick convert - jpgFolder/$file.jpg
> done

Greetings
